gallery/plot_datapoints.py

```diff
@@ -0,0 +1,122 @@
+assert image.data_ptr() == tensor.data_ptr()
+
+
+########################################################################################################################
+# They are needed in :mod:`torchvision.transforms.v2` for the automatic dispatch to the correct kernel.
+#
+# What datapoints are supported?
+# ------------------------------
+#
+# So far :mod:`torchvision.datapoints` supports four different datapoints:
+#
+# * :class:`~torchvision.datapoints.Image`
+# * :class:`~torchvision.datapoints.Video`
+# * :class:`~torchvision.datapoints.BoundingBox`
+# * :class:`~torchvision.datapoints.Mask`
+#
+# How do I construct a datapoint?
+# -------------------------------
+#
+# Each datapoint class takes any tensor-like data that can be turned into a :class:`~torch.Tensor`
+
+image = datapoints.Image([[0, 1], [1, 0]])
+print(image)
+
+
+########################################################################################################################
+# Similar to other PyTorch creations ops, the constructor also takes the ``dtype``, ``device``, and ``requires_grad``
+# parameters.
+
+float_image = datapoints.Image([[0, 1], [1, 0]], dtype=torch.float32, requires_grad=True)
+print(float_image)
+
+
+########################################################################################################################
+# In addition, :class:`~torchvision.datapoints.Image` and :class:`~torchvision.datapoints.Mask` also take a
+# :class:`PIL.Image.Image` directly:
+
+image = datapoints.Image(PIL.Image.open("assets/astronaut.jpg"))
+print(image.shape, image.dtype)
+
+########################################################################################################################
+# In general, the datapoints can also store additional metadata that complements the underlying tensor. For example,
+# :class:`~torchvision.datapoints.BoundingBox` stores the format as well as the spatial size of the corresponding image
+# alongside the actual values:
+
+bounding_box = datapoints.BoundingBox(
+    [17, 16, 344, 495], format=datapoints.BoundingBoxFormat.XYXY, spatial_size=image.shape[-2:]
+)
+print(bounding_box)
+
+
+########################################################################################################################
+# Do I have to wrap the output of the datasets myself?
+# ----------------------------------------------------
+#
+# Only if you are using custom datasets. For the builtin ones, you can use
+# :func:`torchvision.datasets.wrap_dataset_for_transforms_v2`. Note that the function also supports subclasses of the
+# builtin datasets. Meaning, if your custom dataset subclasses from a builtin one and the output type is the same, you
+# also don't have to wrap manually.
+#
+# How do the datapoints behave inside a computation?
+# --------------------------------------------------
+#
+# Datapoints look and feel just like regular tensors. Everything that is supported on plain :class:`torch.Tensor`'s
```

Comment:
super nit: the `'s` doesn't denote plural, does it?

```suggestion
# Datapoints look and feel just like regular tensors. Everything that is supported on a plain :class:`torch.Tensor`
```

Reply:
Not sure what you mean here? If you point out that there shouldn't be an apostrophe for plural, I agree. However, IIRC, sphinx doesn't like putting letters directly after an xref. Let me check.

Reply:
yeah it renders poorly which is why we can circumvent the whole thing by using singular form instead.